### PR TITLE
feat: add queries ShowAuthorization and ListAuthorizations

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

Closes: https://github.com/zeta-chain/node/issues/2307

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: x/authority/keeper/grpc_query_authorization_list.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+func (k Keeper) AuthorizationList(c context.Context,
+	req *types.QueryAuthorizationListRequest,
+) (*types.QueryAuthorizationListResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	authorizationList, found := k.GetAuthorizationList(ctx)
+	if !found {
+		return nil, types.ErrAuthorizationListNotFound
```
[the remark this answers is not in the document]
Comment (kingpinXD) [reply x/authority/keeper/grpc_query_authorization_list.go:23]: No, we would initialize the store using the default list. If the store is not initialized, no authorizations are present. 
I don't think we need to return the default list here , as its querying the store

File: proto/zetachain/zetacore/authority/query.proto [hunk 2]
```diff
@@ -20,6 +21,36 @@ service Query {
   rpc ChainInfo(QueryGetChainInfoRequest) returns (QueryGetChainInfoResponse) {
     option (google.api.http).get = "/zeta-chain/authority/chainInfo";
   }
+
+  rpc AuthorizationList(QueryAuthorizationListRequest)
+      returns (QueryAuthorizationListResponse) {
+    option (google.api.http).get = "/zeta-chain/authority/authorizationList";
```
[the remark this answers is not in the document]
Comment (kingpinXD) [reply proto/zetachain/zetacore/authority/query.proto]: 29d3adaa9e09be91c26e57555728b7791703b891

File: x/authority/keeper/grpc_query_authorization_list.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lumtis) [reply x/authority/keeper/grpc_query_authorization_list.go]: The list can always be set with `admin` policy independently of the table, is it correct? In this case, good to me.

File: x/authority/keeper/grpc_query_authorization_list.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	authorizationList, found := k.GetAuthorizationList(ctx)
+	if !found {
+		return nil, status.Error(codes.Internal, types.ErrAuthorizationListNotFound.Error())
+	}
+	authorization, err := authorizationList.GetAuthorizedPolicy(req.MsgUrl)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	return &types.QueryAuthorizationResponse{Authorization: types.Authorization{
+		MsgUrl:           req.MsgUrl,
```
Comment (lumtis) [inline x/authority/keeper/grpc_query_authorization_list.go:50]: Would be great if we can add some more empty lines in these functions

**Review by lumtis — APPROVED** [review]
LGTM
Minor comment

**Pull request merged**